plugins/webp-uploads/hooks.php

```diff
@@ -192,16 +208,19 @@ function webp_uploads_create_sources_property( array $metadata, int $attachment_
 		}
 
 		// The mime type can't be determined.
-		if ( empty( $current_mime ) ) {
+		if ( ! is_string( $current_mime ) || '' === $current_mime ) {
 			continue;
 		}
 
 		// Ensure a `sources` property exists on the existing size.
-		if ( empty( $properties['sources'] ) || ! is_array( $properties['sources'] ) ) {
+		if ( ! isset( $properties['sources'] ) || ! is_array( $properties['sources'] ) ) {
 			$properties['sources'] = array();
 		}
 
-		if ( empty( $properties['sources'][ $current_mime ] ) && isset( $properties['file'] ) ) {
+		if (
+			( ! isset( $properties['sources'][ $current_mime ] ) || ! is_array( $properties['sources'][ $current_mime ] ) || 0 === count( $properties['sources'][ $current_mime ] ) ) &&
```

Comment:
To make this easier to read:

```suggestion
			(
				! isset( $properties['sources'][ $current_mime ] ) ||
				! is_array( $properties['sources'][ $current_mime ] ) ||
				0 === count( $properties['sources'][ $current_mime ] )
			) &&
```